rs/sns/testing/src/sns.rs

```diff
@@ -32,54 +54,268 @@ pub struct TestCanisterInitArgs {
 pub async fn install_test_canister(pocket_ic: &PocketIc, args: TestCanisterInitArgs) -> CanisterId {
     let topology = pocket_ic.topology().await;
     let application_subnet_ids = topology.get_app_subnets();
-    let application_subnet_id = application_subnet_ids
-        .first()
-        .expect("No Application subnet found");
+    let application_subnet_id = application_subnet_ids[0];
     let features = &[];
     let test_canister_wasm =
         Wasm::from_location_specified_by_env_var("sns_testing_canister", features).unwrap();
     install_canister_on_subnet(
         pocket_ic,
-        *application_subnet_id,
+        application_subnet_id,
         Encode!(&args).unwrap(),
         Some(test_canister_wasm),
         vec![ROOT_CANISTER_ID.get()],
     )
     .await
 }
 
-pub async fn create_sns(
+// PocketIC-specific version of 'create_sns' function.
+// Generates all required identities internally
+// Takes the list of IDs of the DApps that will be added to the SNS as an argument.
+//
+// Returns SNS canisters IDs and the dev participant ID.
+pub async fn create_sns_pocket_ic(
     pocket_ic: &PocketIc,
+    dev_participant_id: PrincipalId,
+    treasury_principal_id: PrincipalId,
     dapp_canister_ids: Vec<CanisterId>,
-) -> (Sns, ProposalId) {
-    let sns_proposal_id = "1";
-    let create_service_nervous_system = CreateServiceNervousSystemBuilder::default()
+) -> Sns {
+    let dev_participant = PocketIcAgent::new(pocket_ic, dev_participant_id);
+
+    let swap_treasury_agent = PocketIcAgent::new(pocket_ic, treasury_principal_id);
+    let swap_partipants_agents = (1..20)
```

Comment:
Why always 20?

Reply:
There is no particular reasoning behind this constant. The way the code is currently organized, agents are created earlier than SNS swap parameters are known. In https://github.com/serokell/ic/pull/6, keys for swap participants are also made static. The main motivation is to be able to reproduce the same set of keys, but I guess this can be reorganized to avoid hardcoding the maximum number of swap participants

Reply:
> reorganized to avoid hardcoding the maximum number of swap participants

Yes, I think we should eventually do this (feel free to postpose it till 3rd milestone).

Note that in the old sns-testing solution, we required 100 participants to finalize the swap. That may be an overkill for some scenarios, but still always having a fixed number doesn't seem right. 

Let's discuss how to avoid that.

Feel free to close the thread.

Reply:
> feel free to postpose it till 3rd milestone

https://github.com/serokell/ic/pull/7 considers the actual swap parameters to generate swap participants instead of using the hardcoded number